modules/imgproc/test/test_convhull.cpp

```diff
@@ -1017,6 +1017,71 @@ int CV_MinCircleTest::validate_test_results( int test_case_idx )
     return code;
 }
 
+/****************************************************************************************\
+*                                 MinEnclosingCircle Test 2                              *
+\****************************************************************************************/
+
+class CV_MinCircleTest2 : public CV_BaseShapeDescrTest
+{
+public:
+    CV_MinCircleTest2();
+protected:
+    RNG rng;
+    void run_func(void);
+    int validate_test_results( int test_case_idx );
+    cv::Point2f center;
+    float radius;
+    vector<Point2f> points;
+};
+
+
+CV_MinCircleTest2::CV_MinCircleTest2()
+{
+    rng = ts->get_rng();
+}
+
+
+void CV_MinCircleTest2::run_func()
+{
+    points.clear();
+    center = Point2f(rng.uniform(0.0f, 1000.0f), rng.uniform(0.0f, 1000.0f));
+    radius = rng.uniform(0.0f, 500.0f);
+    int countBound = rng.uniform(1, 30); // number of boundary points
+    float angle = (float)rng.uniform(0.0f, (float)(CV_2PI)), delta = (float)(CV_2PI / (countBound + 1.0f));
+    points.push_back(center + Point2f(radius * cos(angle), radius * sin(angle)));
+    for (int i = 0; i< countBound; ++i)
+    {
+        angle += delta;
+        points.push_back(center + Point2f(radius * cos(angle), radius * sin(angle)));
+    }
+    float radius2 = radius * radius;
+    float deltaX, x, y, upperBoundY;
+    for (int i = 0; i < rng.uniform(0, 30); ++i)
+    {
+        x = rng.uniform(center.x - radius, center.x + radius);
+        deltaX = x - center.x;
+        upperBoundY = sqrt(radius2 - deltaX * deltaX);
+        y = rng.uniform(center.y - upperBoundY, center.y + upperBoundY);
+        points.push_back(Point2f(x, y));
+    }
+}
+
+int CV_MinCircleTest2::validate_test_results( int test_case_idx )
+{
+    float eps = 1.0F;
+    int code = CV_BaseShapeDescrTest::validate_test_results( test_case_idx );
+    Point2f calcCenter;
+    float calcRadius;
+    // Find the minimum area enclosing circle
+    minEnclosingCircle(points, calcCenter, calcRadius);
```

Comment:
While the test designed such way could detect errors IMHO it's more obvious(and at least more common) to perform result evaluation in `run_func()` function and just verify the result quality in `validate_test_results()`.
Also added test reverify once again the generic case though the original issue is related to specific 3-points case.
Could you please add test or tests to verify specific cases with less than 4 points?

Reply:
Ok, I will edit this. But I didn't understand. Do you want me to change this test so it works only for 3 points or less, or do you want me to add one more?

Reply:
And why do we need to test the case with 2 and 1 points? Maybe it's worth the test for only 3?